packages/web-components/fast-foundation/src/design-token/custom-property-manager.ts

```diff
@@ -17,25 +20,64 @@ interface PropertyTarget {
     removeProperty(name: string);
 }
 
+abstract class QueuedStyleSheetTarget implements PropertyTarget {
+    protected abstract target: PropertyTarget;
+
+    public setProperty(name: string, value: string) {
+        DOM.queueUpdate(() => this.target.setProperty(name, value));
+    }
+    public removeProperty(name: string) {
+        DOM.queueUpdate(() => this.target.removeProperty(name));
+    }
+}
 /**
  * Handles setting properties for a FASTElement using Constructable Stylesheets
  */
-class ConstructableStyleSheetTarget implements PropertyTarget {
-    private target: PropertyTarget;
+class ConstructableStyleSheetTarget extends QueuedStyleSheetTarget {
+    protected target: PropertyTarget;
     constructor(source: HTMLElement & FASTElement) {
+        super();
+
         const sheet = new CSSStyleSheet();
         this.target = (sheet.cssRules[sheet.insertRule(":host{}")] as CSSStyleRule).style;
         source.$fastController.addStyles(ElementStyles.create([sheet]));
     }
+}
 
-    public setProperty(name: string, value: string) {
-        DOM.queueUpdate(() => this.target.setProperty(name, value));
-    }
-    public removeProperty(name: string) {
-        DOM.queueUpdate(() => this.target.removeProperty(name));
+class DocumentStyleSheetTarget extends QueuedStyleSheetTarget {
+    protected target: PropertyTarget;
+    constructor() {
+        super();
+
+        const sheet = new CSSStyleSheet();
+        this.target = (sheet.cssRules[sheet.insertRule(":root{}")] as CSSStyleRule).style;
+        (document as any).adoptedStyleSheets = [
+            ...(document as any).adoptedStyleSheets,
+            sheet,
+        ];
     }
 }
 
+class HeadStyleElementStyleSheetTarget extends QueuedStyleSheetTarget {
+    protected target: PropertyTarget;
+    private readonly style: HTMLStyleElement;
+
+    constructor() {
+        super();
+
+        this.style = document.createElement("style") as HTMLStyleElement;
+        document.head.appendChild(this.style);
```

Comment:
I'm looking for feedback on this point in particular. It feels a little clunky to me to just shove this element into the document head but maybe I'm overthinking it. I'd *like* a way to provide an insertion-point to make this configurable, but right now `.withDefault()` will emit to a CSS custom property immediately so this work will happen prior to an author getting an opportunity to configure DesignToken in cases where a library is defining the default. In this case I suppose we could *move* the node, and maybe that is acceptable? The options as I see them are

1. Determine this is not a problem and leave this as-is
2. Implement some configuration API to provide an insertion-point for this stylesheet. For now we would move the sheet if it is already created, but in the future we could hold emitting default values until some `DesignToken.init()` API is invoked.
3. Look for a magic comment-node like `<!-- fast-design-token-target -->` and emit this element immediately after that node.
4. 2 & 3

Reply:
I think we could go with this as the first version to fix the bug. Then we can follow up with an API to enable declaring the insertion point. It seems like we can probably call that API and mirror it from the DesignSystem. At that point we could delay inserting until component registration, or just insert it and move it. I think I'd avoid magic comments for now. I'd prefer the explicit API. If we need a comment as an anchor point to enable that, I'm fine with that as an internal implementation thing.

Reply:
Yea sounds good. We can add a mechanism to the feature discussed in #4976